GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs

```diff
@@ -67,6 +69,35 @@ public void AddSettingsPage(ISettingsPage page, SettingsPageReference parentPage
             _settingsPages.Add(page);
         }
 
+        private TreeNode AddPage(TreeNodeCollection treeNodeCollection, ISettingsPage page, Bitmap icon)
+        {
+            int imageIndex = GetImageIndex(icon);
+            var node = treeNodeCollection.Add(page.GetTitle());
+            node.ImageIndex = node.SelectedImageIndex = imageIndex;
+            return node;
+        }
+
+        private int GetImageIndex(Bitmap icon)
+        {
+            if (icon == null)
+            {
+                // 0 -> Images.Blank
+                // not -1 because because TreeView displays ImageList[0] when ImageIndex == -1
+                return 0;
+            }
+
+            int imageIndex = treeView1.ImageList.Images.Cast<Bitmap>()
```

Comment:
is this check really needed? this method is called for each node once when the dialog is created.

Reply:
Actually I see you are right. Even though some tree nodes have the same Icon, there is no harm in duplicating them in ImageList, as we are only duplicating the reference to a bitmap, not the bitmap itself.

Will be fixed.